### PR TITLE
Resolve v2/awards API contract

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Resolves discrepancies between frontend/backend and cleans up the API contract.

**Technical details:**

Also updates affected models, components, data mappings, and tests. 

**JIRA Ticket:**
[DEV-2044](https://federal-spending-transparency.atlassian.net/browse/DEV-2044)

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] API contract approved by @tony-sappe 
- [x] [API #1671](https://github.com/fedspendingtransparency/usaspending-api/pull/1671) merged

### CODE REVIEW

**Comment by ebdabbs** [comment]
### Changes for API contract/Frontend
- [Add] `date_signed` to all responses
- [Add] `latest_transaction_contract_data.dod_acquisition_program_description`
- [Allow null] `funding_agency`
- [Allow null] `total_subaward_amount`
- [Change name] `recipient.recipient_parent_name` -> `recipient.parent_recipient_name`
- [Change type to boolean (not NULL)] `small_business_competitive`
- [Change type to string] `latest_transaction_contract_data.number_of_offers_received`
- [Change type to string] `recipient.parent_recipient_unique_id`
- [Change type to string] `recipient.recipient_unique_id`
- [Change/choose name] `business_categories_name` -> `business_categories` for consistency
- [New (not NULL)] `recipient.recipient_hash`
- [Remove] `program_system_or_equipmen` (use `dod_acquisition_program_code`)
- [Rename] `base_and_all_options_value` -> `base_and_all_options` and remove from FinancialAssistanceResponse
- [Rename] `base_exercised_options_val` -> `base_exercised_options`
- [Rename] `federal_action_obligation` -> `total_obligation` for FinancialAssistanceResponse
- [Rename] `funding_obligated` -> `transaction_obligated_amount`
- [Rename] `information_technolog_desc` -> `information_technology_commercial_item_category`
- [Rename] `information_technology_commercial_item` -> `information_technology_commercial_item_category_code`
- [Rename] `latest_transaction_contract_data.dod_aquisition_program` -> `dod_acquisition_program_code`


### New objects to replace PerformancePeriod
1. FPDS (Contract & IDV)
- `end_date` (from `period_of_performance.current_end_date` for contracts, `ordering_period_end_date` for IDV) - string
- `state_date` (`period_of_performance.start_date`) - string
- `potential_end_date` (`period_of_performance.potential_end_date` contracts only?) - string, null
- `last_modified_date` - string

2. Assistance
- `state_date` (`period_of_performance.start_date`) - string
- `end_date` (from `period_of_performance.current_end_date` - string
- `last_modified_date` - string

File: api_contracts/contracts/award/AwardProfile.md [hunk 1]
```diff
@@ -43,61 +43,127 @@ This endpoint returns a list of data that is associated with the award profile p
...
++ type_description: `Definitive Contracts` (required, string)
 + generated_unique_award_id: `25764264` (required, string)
-+ type_description: `Definitive Contract` (required, string)
 + piid: `W31P4Q15A0024` (required, string)
     Award id
 + parent_award_piid: `1301` (required, string, nullable)
+    null if the contract has no parent
 + description: `ewraijwrw` (required, string, nullable)
     Description of the first transaction for this award
++ transaction_obligated_amount: 10700 (required, number, nullable)
```
Comment (tony-sappe) [inline api_contracts/contracts/award/AwardProfile.md]: I don't believe `transaction_obligated_amount` is useful for contracts

File: api_contracts/contracts/award/AwardProfile.md [hunk 2]
```diff
@@ -43,61 +43,127 @@ This endpoint returns a list of data that is associated with the award profile p
...
+    null if the IDV has no parent
 + description: `ewraijwrw` (required, string, nullable)
     Description of the first transaction for this award
++ total_obligation: 30400 (required, number, nullable)
++ base_exercised_options: 23000 (required, number, nullable)
++ base_and_all_options: 24500 (required, number)
+    The ceiling and amount of the award
++ date_signed: `2006-01-15` (required, string, nullable)
++ subaward_count: 430 (required, number)
++ total_subaward_amount: 35345353453 (required, number)
```
Comment (tony-sappe) [inline api_contracts/contracts/award/AwardProfile.md]: can be null

File: api_contracts/contracts/award/AwardProfile.md [hunk 3]
```diff
@@ -123,29 +189,28 @@ This endpoint returns a list of data that is associated with the award profile p
...
++ end_date: `2005-02-19` (required, string)
+    Corresponds to database fields current_end_date for contracts and ordering_period_end_date for IDVs
 + last_modified_date: `2301-02-20` (required, string)
-+ potential_end_date: `2301-02-23` (required, string)
-+ ordering_period_end_date: `2301-02-23` (required, string, nullable)
++ potential_end_date: `2301-02-23` (required, string, nullable)
 
-## IdvDates
+##PeriodOfPerformanceAssistance
 + start_date: `2004-02-19` (required, string)
```
Comment (tony-sappe) [inline api_contracts/contracts/award/AwardProfile.md]: I need to investigate if these fields for Assistance-types can be null, first glance is "yes" (Different than FPDS which appear to be required in upstream systems)

File: api_contracts/contracts/award/AwardProfile.md [hunk 4]
```diff
@@ -43,61 +43,127 @@ This endpoint returns a list of data that is associated with the award profile p
 # Data Structures
 
 ## ContractResponse (object)
-+ type: `A` (required, string)
 + category: `contract` (required, enum[string])
     + Members
         + contract
++ type: `D` (required, enum[string])
```
Comment (tony-sappe) [inline api_contracts/contracts/award/AwardProfile.md]: Nice! I was thinking we should use more enum[string] types.

File: api_contracts/contracts/award/AwardProfile.md [hunk 5]
```diff
@@ -43,61 +43,127 @@ This endpoint returns a list of data that is associated with the award profile p
...
 + description: `ewraijwrw` (required, string, nullable)
     Description of the first transaction for this award
++ transaction_obligated_amount: 10700 (required, number, nullable)
++ total_obligation: 30400 (required, number)
++ base_exercised_options: 23000 (required, number, nullable)
++ base_and_all_options: 24500 (required, number)
+    The ceiling and amount of the award
++ date_signed: `2006-01-15` (required, string, nullable)
++ subaward_count: 430 (required, number)
++ total_subaward_amount: 35345353453 (required, number)
```
Comment (tony-sappe) [inline api_contracts/contracts/award/AwardProfile.md]: Needs to be nullable

**Pull request merged**